rides-and-fares/src/main/java/org/apache/flink/training/exercises/ridesandfares/RidesAndFaresExercise.java

```diff
@@ -98,20 +106,52 @@ public static void main(String[] args) throws Exception {
 
     public static class EnrichmentFunction
             extends RichCoFlatMapFunction<TaxiRide, TaxiFare, RideAndFare> {
-
+        ValueState<TaxiRide> rideState;
+        ValueState<TaxiFare> fareState;
         @Override
         public void open(Configuration config) throws Exception {
-            throw new MissingSolutionException();
+            //Initialize rideState
+            TypeInformation<TaxiRide> rideInfo = TypeInformation.of(new TypeHint<TaxiRide>() {});
+            ValueStateDescriptor<TaxiRide> rideValueStateDescriptor = new ValueStateDescriptor<TaxiRide>("rideState", rideInfo);
+            rideState = getRuntimeContext().getState(rideValueStateDescriptor);
+
+            //Initialize fareState
+            TypeInformation<TaxiFare> fareInfo = TypeInformation.of(new TypeHint<TaxiFare>() {});
+            ValueStateDescriptor<TaxiFare> fareValueStateDescriptor = new ValueStateDescriptor<TaxiFare>("fareState", fareInfo);
+            fareState = getRuntimeContext().getState(fareValueStateDescriptor);
         }
 
         @Override
         public void flatMap1(TaxiRide ride, Collector<RideAndFare> out) throws Exception {
-            throw new MissingSolutionException();
+
+            //Checking if fareState contains the current ride event
+            if(ride!=null && fareState.value()!=null && ride.rideId == fareState.value().rideId){
```

Comment:
Bad code formatting. Please use default IntelliJ IDEA settings for code formatting.

Reply:
@ness-VineetSingh usually you have to reply on this message. Could be something like this "fixed", "done" or ask questions if needed.

Reply:
Done